### ISSUE TITLE
Cannot simultaneously snapshot multiple children nodes in RDataFrame

### ISSUE THREAD

**Issue by NJManganelli** [root-project/root#7500]
- [X] Checked for duplicates

### Describe the bug
If attempting to create multiple snapshots simultaneously of some RDF's children nodes, ROOT crashes. This follows up on a discussion in the thread here:
https://root-forum.cern.ch/t/snapshot-shuffles-data-invisibly-with-implicitmt/43473/8?u=nmangane

### Expected behavior

Expect that one event loop can be triggered with different nodes (downstream of filters, whether or not they produce exclusive sets of events) writing output to unique files via Snapshot. Without this capability, have to fall back to multiple event loops over extensive amounts of data to separate them into different files, which is not really expected.

### To Reproduce

embedded in python script here; I first saw this issue in a more pythonic script (e.g. snapshot options set via cppyy rather than doing everything in the intepreter)
```
import ROOT
ROOT.gInterpreter.ProcessLine("""
ROOT::EnableImplicitMT(8);
auto df = ROOT::RDataFrame(10000).Define("x", "rdfentry_").Define("y", "fabs(rdfentry_ - 5000)").Define("z", "rdfslot_");
ROOT::RDF::RSnapshotOptions opts;
opts.fLazy = true;
auto colNames = df.GetColumnNames();
auto c = df.Count();
auto n1 = df.Filter("x > 2000 && y < 0").Snapshot("Events", "A.root", colNames, opts);
auto n4 = df.Filter("z < 3 && y >= 0").Snapshot("Events", "D.root", colNames);
std::cout << c.GetValue() << std::endl;
""")
```

### Setup

Any recent root version should behave the same. This was first seen in 6.18, 6.20... now confirmed still present in 6.25 (.../dev3/Sun/... build), as obtained via the LCG stack.
```
export SCRAM_ARCH=slc7_amd64_gcc830
source /cvmfs/sft.cern.ch/lcg/views/dev3/Sun/x86_64-centos7-gcc8-opt/setup.sh
```
Reproduced on lxplus (CC7)

**Comment by eguiraud** [comment]
thank you for taking the time to open the issue with a repro, I will take a look as soon as possible

**Comment by eguiraud** [comment]
I think what you are seeing is #7547, which is very real and it will be soon fixed by #7550 .
That problem does not have to do with multiple simultaneous Snapshots however. I will ping you here when we have nightly builds with #7547 fixed so you can check whether you still see problems with simultaneous Snapshots, is that ok?

**Comment by NJManganelli** [comment]
Ah! I had not fully reduced the problem. Thanks!

**Comment by eguiraud** [comment]
@NJManganelli tomorrow's [nightlies](https://root.cern/nightlies) will contain the fix for #7547, please let me know if you still see a crash.

**Comment by NJManganelli** [comment]
Hi @eguiraud, thanks, I've set a reminder to do this tomorrow/early Friday

**Comment by NJManganelli** [comment]
@eguiraud  This works (Mar 18 nightly /dev3/, trying in the real analysis code, where I booked 3 snapshots lazily, then a non-lazy one to start the event loop; to confirm explicitly, the files look good and have reasonable contents. Thanks for the fast turnaround on this! I suppose it'll be a while before this is in a stable LCG release. Not sure if there's a good way to source whatever the latest stable LCG is and then pull in a slightly later but not-nightly root version, but if so that would be great to know about.

Otherwise, thanks again, I can close the issue if this wraps up everything to your satisfaction as well.

**Comment by eguiraud** [comment]
Hey, the fix will be part of 6.24 which is coming out soon
